services/src/main/java/com/chatbot/services/DialogflowConversation.java

```diff
@@ -0,0 +1,123 @@
+package com.chatbot.services;
+
+import com.google.cloud.dialogflow.v2.Context;
+import com.google.cloud.dialogflow.v2.ContextsClient;
+import com.google.cloud.dialogflow.v2.CreateContextRequest;
+import com.google.cloud.dialogflow.v2.DetectIntentRequest;
+import com.google.cloud.dialogflow.v2.DetectIntentResponse;
+import com.google.cloud.dialogflow.v2.QueryInput;
+import com.google.cloud.dialogflow.v2.QueryParameters;
+import com.google.cloud.dialogflow.v2.QueryResult;
+import com.google.cloud.dialogflow.v2.SessionName;
+import com.google.cloud.dialogflow.v2.SessionsClient;
+import com.google.cloud.dialogflow.v2.TextInput;
+import com.google.protobuf.Struct;
+import com.google.cloud.dialogflow.v2.EventInput;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+public class DialogflowConversation {
+
+  private static String projectID;
+  private final String langCode;
+  private final String sessionID;
+  private static Map<String, List<String>> EVENT_TO_CONTEXT_MAPPING = Map.of(
+      "SUGGEST_CATEGORY_CHANGE", Arrays.asList("SuggestChangeCatgeoryContext"),
+      "SUGGEST_IMAGE_UPLOAD", Arrays.asList("SuggestImageUploadContext"),
+      "GET_CALL_FEEDBACK", Arrays.asList("GetCallFeedbackContext")); 
```

Comment:
Might be better to get this from a file. WDYT?

Reply:
updated